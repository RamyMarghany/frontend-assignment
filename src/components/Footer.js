import React from 'react'

import item from '../assets/images/footer-icon.png'


const Footer = () => (
  <div className="row footer-container">
    <div className="col-md-4 col-xs-12">
      <div className="footer">
          <div className="footer__border">
            <img src={item} className="footer__img img-responsive" alt="iphone6"/>
          </div>
          <p className="footer__name">iPhone 6 Black</p>
          <p className="footer__price">aed 1000</p>
        </div>
    </div>
    <div className="col-md-4 col-xs-12">
      <div className="footer">
          <div className="footer__border">
            <img src={item} className="footer__img img-responsive" alt="iphone7"/>
          </div>
          <p className="footer__name">iPhone 7 Black</p>
          <p className="footer__price">aed 2000</p>
        </div>
    </div>
    <div className="col-md-4 col-xs-12">
      <div className="footer">
          <div className="footer__border">
            <img src={item} className="footer__img img-responsive" alt="iphone8"/>
          </div>
          <p className="footer__name">iPhone 8 Black</p>
          <p className="footer__price">aed 3000</p>
        </div>
    </div>
  </div>
)


export default Footer;
