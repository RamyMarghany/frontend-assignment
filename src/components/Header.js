import React, {useState} from 'react'
import {Icon} from 'antd'

import logo from '../assets/images/carrefour.svg'

const Header = () => {
  const [displaySearch, setDisplaySearch] = useState(false);

  return (
    <div className="header">
    <div className="row">
      <div className="col-sm-6 col-xs-12">   
        <img src = {logo} className="header__logo" alt="carrefour"/>
      </div>
      <div className="col-sm-6 col-xs-12">
        <div className="header__search float-sm-right">
          {displaySearch &&
            <>
              <input type="text" className="header__input"/>
              <Icon type="close" className="header__icon" onClick={() => setDisplaySearch(!displaySearch)}/>
            </>
          }
          {!displaySearch && <Icon type="search" className="header__icon" onClick={() => setDisplaySearch(!displaySearch)}/>}
          </div>
       </div>
      </div>
    </div>
  )
}

export default Header