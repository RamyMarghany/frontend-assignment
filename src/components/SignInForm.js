import React, { Component } from 'react'
import {Form, Icon, Input, Button, Checkbox, Col, Row} from 'antd';

class SignInForm extends Component { 
    handleSubmit = (e) => {
        e.preventDefault();
        this.props.form.validateFields((err, values) => {
          if (!err) {
            const data = {
              userName: values.userName,
              password: values.password
            }
            // just for example
            fetch("https://reqres.in/api/users",
            {
                method: "POST",
                body: JSON.stringify(data)
            })
            .then(res => res.json() )
            .then(data => {( console.log( data ))})
            .catch(err => console.log(err))
          }
        });
      }


  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <div className="row sign-in-form">
        <div className="col-xs-10 offset-xs-1 col-md-8 offset-md-2 col-lg-6 offset-lg-3">
          <Form onSubmit={this.handleSubmit} className="login-form">
            <h2 className="login-form__title"> sign in now</h2>
            <p className="login-form__description">Unlock awesome features!</p>
            <Form.Item>
              {getFieldDecorator('userName', {
                rules: [{ required: true, message: 'Required field!' }],
              })(
                <Input  prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />} placeholder="Username" className="login-form__input" />
              )}
            </Form.Item>
            <Form.Item>
              {getFieldDecorator('password', {
                rules: [{ required: true, message: 'Required field!' }],
              })(
                <Input prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />} type="password" placeholder="Password" className="login-form__input" />
              )}
            </Form.Item>
            <Form.Item>
            <Row>
              <Col sm={{ span: 12}} xs={{span:24}} className="align-left">
                {getFieldDecorator('remember', {
                  valuePropName: 'checked',
                  initialValue: true,
                })(
                  <Checkbox>Remember me</Checkbox>
                )}
              </Col>
              <Col sm={{ span: 12}} xs={{span:24}} className="align-right">
                <a className="login-form__forgot">Forgot password</a>
              </Col>
              </Row>
              <Row>
              <Col xs={{ span: 24}}>
                <Button type="primary" htmlType="submit" className="login-form__submit">
                  sign in
                </Button>
              </Col>
              </Row>
            </Form.Item>
          </Form>
        </div>
      </div>
    );
  }
}

export default Form.create({ name: 'normal_login' })(SignInForm);
