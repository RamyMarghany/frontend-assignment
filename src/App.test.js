import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';

import SignInForm from './components/SignInForm';
import Header from './components/Header';
import Footer from './components/Footer';


// test cases for Header
describe('<Header/>', () =>{
  const wrapper = shallow(<Header/>);
  it('should be Logo to be exist',() =>{
    expect(wrapper.find('.header__logo').exists()).toBe(true);
  })

 it('should be search-box to disappear firstly',() =>{
  expect(wrapper.find('.header__input').exists()).toBe(false);
  })

})


// test cases for footer
describe('<Footer/>', () =>{
  const wrapper = shallow(<Footer/>);
  it('should be three footer blocks',() =>{
    expect(wrapper.find('.footer').length).toBe(3);
  })

  it('should be item image to be exist',() =>{
    expect(wrapper.find('.footer__img').exists()).toBe(true);
  })

  it('should be item image to be exist',() =>{
    expect(wrapper.find('.footer__name').exists()).toBe(true);
  })

  it('should be item image to be exist',() =>{
    expect(wrapper.find('.footer__price').exists()).toBe(true);
  })

})


// test cases for sign in form
describe('<SignInForm>',()=>{
  it('matches the snapshot',()=>{
    const tree= shallow(<Footer/>);
    expect(toJson(tree)).toMatchSnapshot();
  })
})
