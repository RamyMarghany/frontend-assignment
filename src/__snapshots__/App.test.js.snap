// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm> matches the snapshot 1`] = `
<div
  className="row footer-container"
>
  <div
    className="col-md-4 col-xs-12 footer"
  >
    <div
      className="footer__border"
    >
      <img
        alt="iphone6"
        className="footer__img img-responsive"
        src="footer-icon.png"
      />
    </div>
    <p
      className="footer__name"
    >
      iPhone 6 Black
    </p>
    <p
      className="footer__price"
    >
      aed 1000
    </p>
  </div>
  <div
    className="col-md-4 col-xs-12 footer"
  >
    <div
      className="footer__border"
    >
      <img
        alt="iphone7"
        className="footer__img img-responsive"
        src="footer-icon.png"
      />
    </div>
    <p
      className="footer__name"
    >
      iPhone 7 Black
    </p>
    <p
      className="footer__price"
    >
      aed 2000
    </p>
  </div>
  <div
    className="col-md-4 col-xs-12 footer"
  >
    <div
      className="footer__border"
    >
      <img
        alt="iphone8"
        className="footer__img img-responsive"
        src="footer-icon.png"
      />
    </div>
    <p
      className="footer__name"
    >
      iPhone 8 Black
    </p>
    <p
      className="footer__price"
    >
      aed 3000
    </p>
  </div>
</div>
`;
